go/vt/vttablet/tabletmanager/rpc_backup.go

```diff
@@ -32,11 +32,6 @@ import (
 
 // Backup takes a db backup and sends it to the BackupStorage
 func (agent *ActionAgent) Backup(ctx context.Context, concurrency int, logger logutil.Logger, allowMaster bool) error {
-	if err := agent.lock(ctx); err != nil {
```

Comment:
Just had a thought: Should we have a separate lock to allow only one backup at a time? I guess technically it might work to have multiple xtrabackups running, but it's probably not a good idea?

Reply:
Would not locking allow it to be promoted to master? Is that ok?

Reply:
+1 for cluster backup lock. It is unlikely you would want a new backup if one is already running, and limiting it helps ensure service only degrades so much.

Reply:
Hm as far as the tablet is concerned, I think that would be acceptable. It's better than blocking promotion to master because xtrabackup is running; presumably you are promoting this one because the current master is in bad shape anyway. WDYT?

I don't know for sure if XtraBackup will still produce a consistent snapshot in that case, but I would expect it to.

Reply:
I don't know a specific reason for it to not be promoted to master, just bringing it up for discussion. Would the tablet type still read `BACKUP`? My guess is that would prevent most workloads from choosing it to failover to, even though they probably should like you said.

Reply:
tablet type cannot change to `BACKUP` if we want the tablet to keep serving while xtrabackup is running.

Reply:
That makes sense. Will there be any knowledge that a backup is running? I would imagine that when selecting a tablet for master promotion, you'd want to prefer a replica not currently running a backup. I would hope to get that info in `wr.ShardReplicationStatuses` or something like it.

Reply:
there should be, but I don't think there is now. Another thing we should fix.